docs/src/main/sphinx/security/jwt-auth.rst

```diff
@@ -0,0 +1,145 @@
+    described on the other system's pages.
+
+Using JWT authentication
+------------------------
+
+A JWT can be passed between servers either in text form or encrypted with a
+standard encryption method described in `RFC 7519
+<https://datatracker.ietf.org/doc/html/rfc7519>`_.
+
+There are two ways to validate a JWT:
+
+-  Acquire its public key from a JSON web key set (JWKS) endpoint service (the
+   typical case)
+-  Extract the signature from the token and validate using a public key
+
+A JWKS endpoint is a read-only service that contains public key information in
+`JWK <https://datatracker.ietf.org/doc/html/rfc7517>`_ format. These public
+keys are the counterpart of the private keys embedded in JSON web tokens.
+
+JWT authentication configuration
+--------------------------------
+
+Do not configure JWT authentication unless you are an administrator trying to
+secure Trino with a custom single sign-on or proxy solution. If that describes
+you, then enable JWT authentication by setting the :doc:`JWT authentication type
+<authentication-types>` in ``etc/config.properties``:
+
+.. code-block:: properties
+
+    http-server.authentication.type=JWT
+
+In addition, you must configure the ``http-server.authentication.jwt.key-file``
+property in ``etc/config.properties``. For example:
+
+.. code-block:: text
+
+    http-server.authentication.jwt.key-file=https://cluster.example.net/.well-known/jwks.json
+
+The following configuration properties for ``etc/config.properties`` are
+available:
+
+.. list-table:: Configuration properties for JWT authentication
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - ``http-server.authentication.jwt.key-file``
+     - Required. Specifies either the URL to a JWKS service or the path to a
+       PEM or HMAC file, as described below this table.
+   * - ``http-server.authentication.jwt.required-issuer``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``iss`` field to consider this JWT valid for this Trino server.
+       The ``iss`` field in the JWT identifies the principal that issued the
+       JWT.
+   * - ``http-server.authentication.jwt.required-audience``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``aud`` field to consider this JWT valid for this Trino server.
+       The ``aud`` field in the JWT identifies the recipients that the JWT is
+       intended for.
+   * - ``http-server.authentication.jwt.principal-field``
+     - Optional. String to identify the field in the JWT that identifies the
+       subject of the JWT. The default value is ``sub``.
+   * - ``http-server.authentication.jwt.user-mapping.pattern``
+     - Optional. A regular expression pattern to map all user names for this
+       authentication system to the format expected by the Trino server. See
+       :doc:`User mapping </security/user-mapping>`.
```

Comment:
change the link to be part of the earlier sentence as descriptive link. Something like

```
A regular expression pattern to the :doc:`define the user mapping </security/user-mapping>` ...
```

Reply:
Done